File: BM5113 - Bilgisayarla Görme/2022/Referance Scripts/lecture6/scale_space.py
from PIL import Image
import numpy as np
import matplotlib.pyplot as plt
import math
import scipy.ndimage as snd

im = np.array(Image.open('./images/jpg/trailer.jpg').convert('L'))

fig, ax = plt.subplots(nrows=1, ncols=4)
plt.gray()

for sig in range(1,11,3): 
    out = snd.filters.gaussian_filter(im,sig)
    indexAxis = math.floor((sig-1)/3)+1
    plt.subplot(1,4,indexAxis), plt.imshow(out)
    plt.title((r'Gauss blur, $\sigma=$'+str(sig)))

plt.show()

File: BM5113 - Bilgisayarla Görme/reference_scripts/matlab functions II/scale_space.m
I=im2double(imread('../images/street.png'));
figure;
close all;
increaseSigma = 5;
maxSigma = 26;
out = zeros(size(I,1),size(I,2),floor(maxSigma/increaseSigma)+1);
for sig = 1:increaseSigma:maxSigma
    indexAxis = floor((sig-1)/increaseSigma)+1;
    out(:,:,indexAxis) = imgaussfilt(I,sig);
    subplot(2,3,indexAxis);
    imshow(out(:,:,indexAxis));
    title(['Gauss blur, sigma=' num2str(sig)])
end
figure;
for i= 1:floor(maxSigma/increaseSigma)
    DoG = out(:,:,i)-out(:,:,i+1);
    subplot(2,3,i),imshow(DoG, [min(DoG(:)) max(DoG(:))]);
end